shell/shared/grain.js

```diff
@@ -590,9 +590,14 @@ if (Meteor.isClient) {
           assignment = call.roleAssignment;
         }
         // Tokens expire by default in 5 minutes from generation date
-        var selfDestructTime = Date.now() + (5 * 60 * 1000);
+        var selfDestructDuration = 5 * 60 * 1000;
+
+        var rawParentToken;
+        if (Router.current().route.getName() === "shared") {
```

Comment:
When the user navigates away from the grain, do we dispose of the `#grain-frame` iframe, or do we simply hide it?  I can see a TOCTOU bug hiding here if the grain frame is hidden, the user navigates elsewhere (changing `Router.current().route`), then the app `postMessage()`s a renderTemplate request.